src/Compilers/CSharp/Portable/CodeGen/EmitExpression.cs

```diff
@@ -352,6 +356,41 @@ private void EmitExpressionCore(BoundExpression expression, bool used)
             }
         }
 
+        private void EmitLoweredIsPatternExpression(BoundLoweredIsPatternExpression node, bool used)
+        {
+            EmitSideEffects(node.Statements);
+
+            if (!used)
+            {
+                _builder.MarkLabel(node.WhenTrueLabel);
+                _builder.MarkLabel(node.WhenFalseLabel);
+            }
+            else
+            {
+                var doneLabel = new object();
+                _builder.MarkLabel(node.WhenTrueLabel);
+                _builder.EmitBoolConstant(true);
+                _builder.EmitBranch(ILOpCode.Br, doneLabel);
+                _builder.AdjustStack(-1);
+                _builder.MarkLabel(node.WhenFalseLabel);
+                _builder.EmitBoolConstant(false);
+                _builder.MarkLabel(doneLabel);
+            }
+        }
+
+        private void EmitSideEffects(ImmutableArray<BoundStatement> statements)
+        {
+#if !DEBUG
```

Comment:
This is rather unconventional. There is only several places where ```#if !DEBUG``` is used in the compiler. All of them do nothing for DEBUG, which is not the case here. Consider adjusting to use ```#if DEBUG``` instead.